src/clib/lib/job_queue/torque_driver.cpp

```diff
@@ -575,74 +581,91 @@ job_status_type torque_driver_parse_status(const char *qstat_file,
+        if (dot_position != std::string::npos) {
+            jobnr_namespaced.replace(dot_position, 1, " ");
         }
+        std::stringstream(jobnr_namespaced) >> jobnr_no_namespace;
+    }
 
-        if (line) {
-            char job_id_full_string[32];
-            char string_status[2];
-
-            if (sscanf(line, "%s %*s %*s %*s %s %*s", job_id_full_string,
-                       string_status) == 2) {
-                /* The job id string may or may not contain a dot */
-                const char *dotPtr = strchr(job_id_full_string, '.');
-                {
-                    char *job_id_as_char_ptr;
-                    if (dotPtr != NULL) {
-                        int dotPosition = dotPtr - job_id_full_string;
-                        job_id_as_char_ptr = util_alloc_substring_copy(
-                            job_id_full_string, 0, dotPosition);
-                    } else {
-                        job_id_as_char_ptr =
-                            util_alloc_string_copy(job_id_full_string);
-                    }
-                    if (util_string_equal(job_id_as_char_ptr, jobnr_char)) {
-
-                        switch (string_status[0]) {
-                        case 'R':
-                            /* Job is running */
-                            status = JOB_QUEUE_RUNNING;
-                            break;
-                        case 'E':
-                            /* Job is exiting after having run */
-                            status = JOB_QUEUE_DONE;
-                            break;
-                        case 'F':
-                            /* PBS specific value: Job is finished */
-                            /* This is only returned in the alternative qstat format
-                            triggered with '-x' or '-H' option to qstat */
-                            status = JOB_QUEUE_DONE;
-                            break;
-                        case 'C':
-                            /* Job is completed after having run */
-                            status = JOB_QUEUE_DONE;
-                            break;
-                        case 'H':
-                            /* Job is held */
-                            status = JOB_QUEUE_PENDING;
-                            break;
-                        case 'Q':
-                            /* Job is queued, eligible to run or routed */
-                            status = JOB_QUEUE_PENDING;
-                            break;
-                        default:
-                            break;
-                        }
-
-                        free(job_id_as_char_ptr);
-                    }
+    /* Parse the qstat output, looking only for requested job_id */
+    if (fs::exists(qstat_file)) {
+
+        std::string job_id_label("Job Id: ");
+        int pos;
+        int job_id = -1;
+        std::string job_state("_void_");
+        int exit_status = 0;
+        std::ifstream qstatoutput(qstat_file);
```

Comment:
These days I like to add:
```
qstatoutput.exceptions(qstatoutput.failbit);
```

This will throw a C++ exception if the file doesn't exist (it'll check and potentially fail as soon as you call `.exceptions(`), if you try to read a formatted value that isn't correct (ie, if the buffer doesn't contain a number and you try to read a number, throw exception). Notably, if you get EOF it doesn't throw (until you try to read again I think).

Handle errors with
```c++
try {
    int number;
    qstatoutput >> number;
} catch (std::ios::failure &) {
    // do something else
}
```

*Also*:
Use `qstatoutput.imbue(std::locale::classic());`. This will set C locale to this stream object and will help us avoid the `,` vs `.` problems that we currently have.

See: https://en.cppreference.com/w/cpp/io/basic_ios/fail